test/unit/cli/parse-options-test.js

```diff
@@ -68,6 +72,57 @@ test('parse options', function (group) {
     t.end()
   })
 
+  group.test('dbUrl', function (t) {
+    var config = parseOptions({
+      public: 'public',
+      data: 'data',
+      dbUrl: 'https://foo:bar@baz.com'
+    })
+    var defaults = config.PouchDB('hack', {skip_setup: true}).__opts
+    t.is(defaults.prefix, 'https://foo:bar@baz.com', 'Sets config.db.url')
+
+    t.end()
+  })
+
+  // https://foo@baz.com passing
+  group.test('dbUrl', function (t) {
+    var config = parseOptions({
+      public: 'public',
+      data: 'data',
+      dbUrl: 'https://https://foo@baz.com'
+    })
+    var defaults = config.PouchDB('hack', {skip_setup: true}).__opts
+    t.is(defaults.prefix, 'https://https:%2F%2Ffoo@baz.com', 'Sets config.db.url')
+
+    t.end()
+  })
+
+  // @@:@@@baz.com passing
+  group.test('dbUrl', function (t) {
+    var config = parseOptions({
+      public: 'public',
+      data: 'data',
+      dbUrl: 'http://@@:@@@baz.com'
+    })
+    var defaults = config.PouchDB('hack', {skip_setup: true}).__opts
+    t.is(defaults.prefix, 'http://%40%40:%40%40@baz.com', 'Sets config.db.url')
+
+    t.end()
+  })
+
+  // dasds@dasdsa.com:@dsadas@dasdas@@baz.com passing
+  group.test('dbUrl', function (t) {
+    var config = parseOptions({
+      public: 'public',
+      data: 'data',
+      dbUrl: 'http://dasds@dasdsa.com:@dsadas@dasdas@@baz.com'
+    })
+    var defaults = config.PouchDB('hack', {skip_setup: true}).__opts
+    t.is(defaults.prefix, 'http://dasds%40dasdsa.com:%40dsadas%40dasdas%40@baz.com', 'Sets config.db.url')
+
+    t.end()
+  })
```

Comment:
For my code, arguments `--dbUrlUsername` and `--dbUrlPassword` have a higher priority than auth details provided in `--dbUrl` (I think it's more logically correct), thus they will override the auth details included in `--dbUrl`. I am going to rewrite the tests, add more of them and describe each one of them clearly